lib/IterateUtils.lua

```diff
@@ -0,0 +1,49 @@
+local IterateUtils = {}
+
+local function getNext(source)
+	if type(source) == "function" then
+		return source
+	else
+		local key = 0
+		return function()
+			key = key + 1
+			if source[key] == nil then
+				return
+			else
+				return key, source[key]
+			end
+		end
+	end
+end
+
+function IterateUtils.getInsertIterator(source)
+	local next = getNext(source)
```

Comment:
assertions

Reply:
Still to do.

Reply:
the validity of source is asserted in getNext, no need for a duplicate assertion